### PR TITLE
chore: readme

### PR DESCRIPTION
**Pull request by tormodAase**

### CODE REVIEW

**Comment by tormodAase** [comment]
We need react-native-gesture-handler 2.13.0 or higher to get access to `SwipeableProps`. While we could find some workarounds temporarily, [Expo 50 is just around the corner](https://expo.dev/changelog/2023/12-12-sdk-50-beta), so I think it's better to just leave `react-native-gesture-handler` at `^2.14.0`. Expo doctor might complain, but I haven't run into any issues

File: packages/core/README.md [hunk 1]
```diff
@@ -1,3 +1,396 @@
...
+```
+
+#### Step 4: replace `createStackNavigator`/`createNativeStackNavigator`/`createBottomTabsNavigator`/`NavigationContainer` imports
+
+In order for us to display environment banners and service messages on all screens, you need to
+import `createNativeStackNavigator`/`createBottomTabsNavigator` from `@equinor/mad-core` instead of
+`react-navigation`. Currently we don't support other navigators. Do you need support for other
+navigators? Create an issue!
+
+By default environment-banners and service-messages should display if the screen header is
```
Comment (dansvend) [inline packages/core/README.md:255]: With dash in `environment-banners` and `service-messages` here and without in the previous paragraph. I'd stick to the spelling in the first paragraph.

File: packages/core/README.md [hunk 2]
```diff
@@ -1,3 +1,396 @@
...
+                name="cog-outline"
+                variant="ghost"
+                onPress={() => navigation.navigate("Settings")}
+            />
+        </View>
+    );
+}
+```
+
+Results: ![[Pasted image 20231221114201.png]]
```
Comment (dansvend) [inline packages/core/README.md:299]: This doesn't work.

**Review by dansvend — CHANGES_REQUESTED** [review]
Really nice guide - thanks for putting this together🙏 I added some improvement suggestions and questions🙌

File: packages/core/README.md [hunk 3]
```diff
@@ -1,3 +1,396 @@
...
+
+###### Language config
+
+| key                   | required? | explanation                                                                                                                                                                                                                                                                                         |
+| --------------------- | --------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `supportedLanguages`  | true      | an array of supported languages. The language object should contain `code` and `name`. `mad-core` supports Norwegian (`no`, `nb`), english (`en`) and Portuguese (`pt`). If you use any other languages, the common screens will be in english. If you need more language support, create an issue! |
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:125]: If Equinor decides to support Nynorsk in the future, we would most likely use `nb` and `nn`. If anything should be removed, it should be `no`, but to make it less complicated, we support both `no` and `nb`

File: packages/core/README.md [hunk 4]
```diff
@@ -1,3 +1,396 @@
 # MAD CORE
 
-This package is not yet ready for use. It is a work in progress
+_`@equinor/mad-core` is a package made for the mad team in equinor. It is opinionated and will not
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:3]: Fixed

File: packages/core/README.md [hunk 5]
```diff
@@ -1,3 +1,396 @@
...
+
+`@equinor/mad-core` is designed to handle the general requirements for your react native app, so
+that you can focus on your app's core functionality. It provides:
+
+-   Authentication system, including a login screen and easy authentication for your needs
+-   Release notes. No need to worry about displaying release-notes. `mad-core` handles everything
+    for you. If the app version is greater than the last time the user used it, a "What's
+    New"-screen will be displayed automatically.
+-   Language support. If you set your app to support multiple languages, the app will prompt the
+    user to select language. Language will also be provided as a setting in the Settings page if
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:17]: Fixed

File: packages/core/README.md [hunk 6]
```diff
@@ -1,3 +1,396 @@
...
+    multiple languages are supported.
+-   Settings Screen. The Settings Screen will provide a lot of your common needs, depending on your
+    config. Your app-specific needs should also be easy to add.
+-   Application Insights tracking. We provide methods for tracking events in your app. We have also
+    added tracking to some of our screens, a `NavigationContainer` you can use to automatically
+    track navigation, and a `ErrorBoundary` you can use for user-friendly crash handling, as well as
+    crash tracking.
+-   Service Message. Any planned maintenance? Just create a service message in the service portal,
+    and the message will be displayed in the app automatically.
+-   Environment banners. Don't know which environment you're in? Don't worry, we will display the
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:27]: Fixed

File: packages/core/README.md [hunk 7]
```diff
@@ -1,3 +1,396 @@
...
+    Amazing!
+-   Create incident screen. For creating a new incident in service-now. Available in a
+    settings-screen near you!
+-   Probably more, but it's hard to keep track of everything. Do you know of anything that should be
+    added here? Create an issue!
+
+### Implementation
+
+_note: this package is new. Expect bugs. Expect lacking documentation. If you have any issues,
+PLEASE create an issue, and yell at @TormodAase. We want to make this great, but it's impossible to
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:39]: Removed the part mentioning me

File: packages/core/README.md [hunk 8]
```diff
@@ -1,3 +1,396 @@
...
+
+#### Step 1: Installation
+
+Install `react-native-msal` from [this repository](https://github.com/equinor/react-native-msal/).
+This is a fork maintained by the mad-team. Always install the newest commit. To install it, add this
+to your `package.json`:
+
+```json
+"dependencies": {
+    "react-native-msal": "git+https://github.com/equinor/react-native-msal.git#NEWEST_COMMIT_HASH_HERE",
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:50]: Fixed

File: packages/core/README.md [hunk 9]
```diff
@@ -1,3 +1,396 @@
...
+```tsx
+import { MadConfig } from "@equinor/mad-core";
+import Logo from "./assets/images/icon.png";
+import { ImageSourcePropType } from "react-native";
+import { getBuildNumber } from "./settings";
+
+export const config: MadConfig = {
+    appVersion: "1.0.0",
+    servicePortalName: "Chronicles",
+    environment: "prod",
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:76]: Fixed

File: packages/core/README.md [hunk 10]
```diff
@@ -1,3 +1,396 @@
...
+        title: "Chronicles",
+        logo: Logo as ImageSourcePropType,
+    },
+    applicationInsights: {
+        instrumentationKey: "f1859360-4aa2-425f-b494-2d7320de6832",
+        longTermLog: { instrumentationKey: "e91835aa-bcc2-41dd-a79d-352f0df23e1b" },
+    },
+    serviceNow: {
+        whatever: "",
+    },
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:101]: Fixed

File: packages/core/README.md [hunk 11]
```diff
@@ -1,3 +1,396 @@
...
+    about: {
+        endpoints: [],
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:103]: Fixed

File: packages/core/README.md [hunk 12]
```diff
@@ -1,3 +1,396 @@
...
+        buildNumber: getBuildNumber(),
+    },
+};
+```
+
+| key                   | required? | explanation                                                                                                                           |
+| --------------------- | --------- | ------------------------------------------------------------------------------------------------------------------------------------- |
+| `appVersion`          | true      | Your app's current version. Used to figure out whether we should display what's new, and which release notes version to fetch         |
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:111]: `we` replaced by `the app`

File: packages/core/README.md (hunk 1 above)
Comment (tormodAase) [reply packages/core/README.md:255]: Fixed

File: packages/core/README.md [hunk 13]
```diff
@@ -1,3 +1,396 @@
...
+        longTermLog: { instrumentationKey: "e91835aa-bcc2-41dd-a79d-352f0df23e1b" },
+    },
+    serviceNow: {
+        whatever: "",
+    },
+    about: {
+        endpoints: [],
+        buildNumber: getBuildNumber(),
+    },
+};
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/core/README.md:106]: Added `chronicles` example

File: packages/core/README.md (hunk 2 above)
Comment (tormodAase) [reply packages/core/README.md]: Fixed

File: packages/core/README.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (dansvend) [reply packages/core/README.md]: Nice one 👌

**Review by dansvend — APPROVED** [review]
This documentation is really helpful. Helped me a ton when migrating to this package 🙏  Thanks for putting this together, @tormodAase - well done 💪

**Pull request merged**